tests/ssl-opt.sh

```diff
@@ -2286,30 +2286,36 @@ run_test    "TLS 1.3: key exchange mode parameter passing: All" \
 requires_openssl_tls1_3
 requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_3
 requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
+requires_config_enabled MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
 requires_config_enabled MBEDTLS_SSL_SRV_C
 requires_config_enabled MBEDTLS_DEBUG_C
-run_test    "TLS 1.3: psk_key_exchange_modes: basic check, O->m" \
-            "$P_SRV force_version=tls13 debug_level=5" \
-            "$O_NEXT_CLI -tls1_3 -psk 6162636465666768696a6b6c6d6e6f70 -allow_no_dhe_kex" \
-            0 \
+run_test    "TLS 1.3: PSK: basic check, O->m" \
+            "$P_SRV force_version=tls13 tls13_kex_modes=psk debug_level=5 psk=6162636465666768696a6b6c6d6e6f70" \
+            "$O_NEXT_CLI -tls1_3 -psk 1234 -psk 6162636465666768696a6b6c6d6e6f70 -allow_no_dhe_kex" \
+            1 \
             -s "found psk key exchange modes extension" \
+            -s "found pre_shared_key extension" \
             -s "Found PSK_EPHEMERAL KEX MODE" \
-            -s "Found PSK KEX MODE"
+            -s "Found PSK KEX MODE" \
+            -s "Pre shared key found"
 
 requires_gnutls_tls1_3
 requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_3
 requires_config_enabled MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
+requires_config_enabled MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
 requires_config_enabled MBEDTLS_SSL_SRV_C
 requires_config_enabled MBEDTLS_DEBUG_C
-run_test    "TLS 1.3: psk_key_exchange_modes: basic check, G->m" \
-            "$P_SRV force_version=tls13 debug_level=5" \
-            "$G_NEXT_CLI --priority NORMAL:-VERS-ALL:+VERS-TLS1.3 \
+run_test    "TLS 1.3: PSK: basic check, G->m" \
+            "$P_SRV force_version=tls13 tls13_kex_modes=psk debug_level=5 psk=6162636465666768696a6b6c6d6e6f70" \
+            "$G_NEXT_CLI --priority NORMAL:-VERS-ALL:+KX-ALL:+PSK:+DHE-PSK:+VERS-TLS1.3 \
                          --pskusername Client_identity --pskkey=6162636465666768696a6b6c6d6e6f70 \
                          localhost" \
-            0 \
+            1 \
```

Comment:
As I said offline, I managed to complete a pure PSK handshake with this PR plus a few additional changes in code not related to this PR: see https://github.com/ronald-cron-arm/mbedtls/tree/pr6079-PSK-handshake-with-GnuTLS. What I have done only works when USE_PSA_CRYPTO is disabled like in TLS 1.3 only configuration. The case where USE_PSA_CRYPTO is defined is yet to be done.